libp2p-floodsub/src/lib.rs

```diff
@@ -0,0 +1,640 @@
+        self.subscribe_multi(iter::once(topic));
+    }
+
+    /// Same as `subscribe`, but subscribes to multiple topics at once.
+    #[inline]
+    pub fn subscribe_multi<'a, I>(&self, topics: I)
+    where
+        I: IntoIterator<Item = &'a Topic>,
+        I::IntoIter: Clone,
+    {
+        // This function exists for convenience.
+        self.sub_unsub_multi(topics.into_iter().map::<_, fn(_) -> _>(|t| (t, true)))
+    }
+
+    /// Unsubscribe from a topic. We will no longer receive any message for this topic.
+    ///
+    /// If a message was sent to us before we are able to notify that we don't want messages
+    /// anymore, then the message will be filtered out locally.
+    #[inline]
+    pub fn unsubscribe(&self, topic: &Topic) {
+        // This function exists for convenience.
+        self.unsubscribe_multi(iter::once(topic));
+    }
+
+    /// Same as `unsubscribe` but unsubscribes from multiple topics at once.
+    #[inline]
+    pub fn unsubscribe_multi<'a, I>(&self, topics: I)
+    where
+        I: IntoIterator<Item = &'a Topic>,
+        I::IntoIter: Clone,
+    {
+        // This function exists for convenience.
+        self.sub_unsub_multi(topics.into_iter().map::<_, fn(_) -> _>(|t| (t, false)));
+    }
+
+    // Inner implementation. The iterator should produce a boolean that is true if we subscribe and
+    // false if we unsubscribe.
+    fn sub_unsub_multi<'a, I>(&self, topics: I)
+    where
+        I: IntoIterator<Item = (&'a Topic, bool)>,
+        I::IntoIter: Clone,
+    {
+        let mut proto = rpc_proto::RPC::new();
+
+        let topics = topics.into_iter();
+
+        debug!(target: "libp2p-floodsub", "Queuing sub/unsub message ; sub = {:?} ; unsub = {:?}",
+               topics.clone().filter(|t| t.1)
+                     .map(|t| t.0.hash().clone().into_string())
+                     .collect::<Vec<_>>(),
+               topics.clone().filter(|t| !t.1)
+                     .map(|t| t.0.hash().clone().into_string())
+                     .collect::<Vec<_>>());
+
+        let mut subscribed_topics = self.inner.subscribed_topics.write();
+        for (topic, subscribe) in topics {
+            let mut subscription = rpc_proto::RPC_SubOpts::new();
+            subscription.set_subscribe(subscribe);
+            subscription.set_topicid(topic.hash().clone().into_string());
+            proto.mut_subscriptions().push(subscription);
+
+            if subscribe {
+                subscribed_topics.push(topic.clone());
+            } else {
+                subscribed_topics.retain(|t| t.hash() != topic.hash())
+            }
+        }
+
+        self.broadcast(proto, |_| true);
+    }
+
+    /// Publishes a message on the network for the specified topic
+    #[inline]
+    pub fn publish(&self, topic: &Topic, data: Vec<u8>) {
+        // This function exists for convenience.
+        self.publish_multi(iter::once(topic), data)
+    }
+
+    /// Publishes a message on the network for the specified topics.
```

Comment:
Same as above, mention that `publish_multi` is preferred to multiple calls to `publish` and consider calling this `publish_many`.